routines.go

```diff
@@ -470,3 +499,165 @@ func WebsocketReconnect(ws *exchange.Websocket, verbose bool) {
 		}
 	}
 }
+
+// PlatformTradeUpdaterRoutine fetches and updates platform trade data and
+// enters said data into defined database
+func PlatformTradeUpdaterRoutine() {
+	log.Debug("Starting platform trade fetching routine..")
+	for {
+		wg.Add(len(bot.exchanges))
+		for x := range bot.exchanges {
+			go func(x int, wg *sync.WaitGroup) {
+				defer wg.Done()
+
+				if bot.exchanges[x] == nil {
+					return
+				}
+
+				if !bot.db.IsConnected() {
+					log.Errorf("Exchange %s platform trade updater failed to fetch, not connected to database",
+						bot.exchanges[x].GetName())
+					return
+				}
+
+				exchangeName := bot.exchanges[x].GetName()
+				dbconfig, err := bot.config.GetDatabaseConfig(exchangeName)
+				if err != nil {
+					log.Errorf("failed to get %s exchange database configuration. Error: %s",
+						exchangeName,
+						err)
+					return
+				}
+
+				if !dbconfig.LoadPlatformTrades {
+					log.Debugf("Exchange %s platform trade updater not enabled in config.json",
+						bot.exchanges[x].GetName())
+					return
+				}
+
+				enabledCurrencies := bot.exchanges[x].GetEnabledCurrencies()
+				assetTypes, err := exchange.GetExchangeAssetTypes(exchangeName)
+				if err != nil {
+					log.Errorf("failed to get %s exchange asset types. Error: %s",
+						exchangeName,
+						err)
+					return
+				}
+
+				for y := range assetTypes {
+					for z := range enabledCurrencies {
+						fetchHistory(bot.exchanges[x],
+							enabledCurrencies[z],
+							assetTypes[y],
+							time.Unix(dbconfig.TimestampStart, 0),
+							time.Unix(dbconfig.TimestampEnd, 0))
+					}
+				}
+			}(x, &wg)
+		}
+		wg.Wait()
+		log.Debugln("All enabled currency platform trades fetched.")
+		time.Sleep(time.Minute * 5)
+	}
+}
+
+func fetchHistory(exch exchange.IBotExchange, p currency.Pair, assetType string, timeStart, timeEnd time.Time) {
```

Comment:
As we discussed breifly yesteray the timeStart and timeEnd usage in fetchHistory is incorrect because GetPlatformHistory for most exchanges will only accept a start time and then each exchange wrapper sets up these times depending on the exchange limits (for example Binance is 3 months)

Also on lines 660-661 you are just printing time.Time{} which will cause the output to always be 1970-01-01 10:00:00

Reply:
Will fix!